Packages/Sequence-Unity/Sequence/SequenceSDK/Marketplace/TransferFundsViaQR.cs

```diff
@@ -0,0 +1,98 @@
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Networking;
+using Sequence;
+using Sequence.EmbeddedWallet;
+using Sequence.Marketplace;
+
+public class TransferFundsViaQR : MonoBehaviour, ICheckoutOption
+{
+    [SerializeField] GameObject _qrPanel;
+    [SerializeField] Image _qrImage;
+
+    SequenceWallet _wallet;
+    CollectibleOrder _order;
+
+    string apiEndpoint = "https://api.qrserver.com/v1/create-qr-code/";
+   
+
+    void Awake()
+    {
+        GetComponent<Button>().onClick.AddListener(() => Checkout());
+    }
+    public async void Checkout()
+    {
+        if (_order != null)
+        {
+            await SetQrCode((int)_order.order.chainId, _wallet.GetWalletAddress(), ConvertToScientificNotation(float.Parse(_order.order.priceAmount)));
+            _qrPanel.SetActive(true);
+        }
+        else Debug.LogError("Collectible order not set for checkout option.");
+    }
+
+    public void SetWallet(SequenceWallet wallet)
+    {
+        _wallet = wallet;
+    }
+
+
+    public void SetCollectibleOrder(CollectibleOrder checkoutOrder)
+    {
+        _order = checkoutOrder;
+    }
+    public async Task SetQrCode(int chainId, string destinationAddress, string amount)
+    {
+        var texture = await GenerateQRCodeAsync(chainId, destinationAddress, amount);
+        _qrImage.sprite = Sprite.Create(texture, new Rect(0, 0, texture.width, texture.height), new UnityEngine.Vector2(0.5f, 0.5f));
+    }
+
+    public async Task<Texture2D> GenerateQRCodeAsync(int chainId, string destinationAddress, string amount)
```

Comment:
This is great! We should also add a more generic QR code that just gives the destination address and ideally the chain id as well - this would be useful for non-metamask users